src/sage/graphs/matching_covered_graph.py

```diff
@@ -0,0 +1,1228 @@
+            sage: H = BipartiteGraph(MatchingCoveredGraph(G))
+            sage: H  # An object of the class BipartiteGraph
+            Bipartite hexahedron: graph on 8 vertices
+            sage: J = MatchingCoveredGraph(H)
+            sage: J  # An object of the class MatchingCoveredGraph
+            Matching covered hexahedron: graph on 8 vertices
+        """
+        s = Graph._repr_(self).lower()
+        if "matching covered" in s:
+            return s.capitalize()
+        return "".join(["Matching covered ", s])
+
+    def add_edge(self, u, v=None, label=None):
+        r"""
+        Add an edge from vertex ``u`` to vertex ``v``.
+
+        .. NOTE::
+
+            This method overwrites the
+            :meth:`~sage.graphs.generic_graph.GenericGraph.add_edge` method
+            to ensure that resultant graph is also matching covered.
+
+        INPUT:
+
+        The following forms are all accepted:
+
+        - G.add_edge(1, 2)
+        - G.add_edge((1, 2))
+        - G.add_edges([(1, 2)])
+        - G.add_edge(1, 2, 'label')
+        - G.add_edge((1, 2, 'label'))
+        - G.add_edges([(1, 2, 'label')])
+
+        OUTPUT:
+
+        - If an edge is provided with a valid format, but addition of the edge
+          leaves the resulting graph not being matching covered, a
+          :exc:`ValueError` is returned without any alteration to the existing
+          matching covered graph. If the addition of the edge preserves the
+          property of matching covered, then the graph is updated and nothing
+          is returned.
+
+        - If the edge is provided with an invalid format, a :exc:`ValueError`
+          is returned.
+
+        WARNING:
+
+        The following intuitive input results in nonintuitive output,
+        even though the resulting graph behind the intuition might be matching
+        covered::
+
+            sage: P = graphs.WheelGraph(6)
+            sage: G = MatchingCoveredGraph(P)
+            sage: G.add_edge((1, 4), 'label')
+            Traceback (most recent call last):
+            ...
+            ValueError: the graph obtained after the addition of edge (((1, 4), 'label', None)) is not matching covered
+            sage: G.edges(sort=False)
+            [(0, 1, None), (0, 2, None), (0, 3, None), (0, 4, None), (0, 5, None), (1, 2, None), (1, 5, None), (2, 3, None), (3, 4, None), (4, 5, None)]
+
+        The key word ``label`` must be used::
+
+            sage: W = graphs.WheelGraph(6)
+            sage: G = MatchingCoveredGraph(W)
+            sage: G.add_edge((1, 4), label='label')
+            sage: G.edges(sort=False)  # No alteration to the existing graph
+            [(0, 1, None), (0, 2, None), (0, 3, None), (0, 4, None), (0, 5, None), (1, 2, None), (1, 4, 'label'), (1, 5, None), (2, 3, None), (3, 4, None), (4, 5, None)]
+
+        An expression, analogous to the syntax mentioned above may be used::
+
+            sage: W = graphs.WheelGraph(6)
```

Comment:
Well, you can certainly combine all these tests to show the different ways of adding an edge, for instance first adding edge (1, 4), then another edge connecting different vertices (possibly adding new vertices to the graph, etc.